node/src/accountant/db_access_objects/failed_payable_dao.rs

```diff
@@ -21,11 +23,83 @@ pub enum FailedPayableDaoError {
     SqlExecutionFailed(String),
 }
 
-#[derive(Clone, Debug, PartialEq, Eq)]
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
 pub enum FailureReason {
+    Local(LocalError),
+    Api(ApiError),
+    Pool(PoolError),
```

Comment:
We've discussed this on Discord already.

Please introduce this distinguishing for the Api variant. 

```Rust

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
pub struct ApiError {
     pub phase: TxPhase, 
     pub msg: String
}


#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
pub enum TxPhase {
    Sending,
    ReceiptCheck
}
```

or if you prefer, you can also add the two fields straight into the Api variant. With the struct-like syntax. 

```Rust
Api {
    phase: TxPhase,
    msg: String
}
```